examples/linear_deformation.py

```diff
@@ -0,0 +1,1187 @@
+            The points ``x_j`` controlling the deformation. They can
+            be given either as a tensor grid or as a point array. In
+            the latter case, its shape must be ``(N, n)``, where
+            ``n`` is the dimension of the template space, and ``N``
+            the number of ``alpha_j``, i.e. the size of (each
+            component of) ``par_space``.
+        discr_space : `DiscreteSpace`
+            Space of the image grid of the template.
+        kernel : `callable`
+            Function to determine the kernel at the control points ``K(y_j)``
+            The function must accept a real variable and return a real number.
+        """
+
+        super().__init__(alphas, control_points, discr_space, ft_kernel)
+
+        # Switch domain and range
+        self._domain, self._range = self._range, self._domain
+
+    def _call(self, grad_func):
+        """Implement ``self(func)```.
+
+        Parameters
+        ----------
+        func : `DiscreteLpVector`
+            Element of the image's gradient space.
+        """
+        # If control grid is not the image grid, the following result for
+        # the ajoint is not right. Because the kernel matrix in fitting
+        # term is not symmetric.
+        return self.displacement_ft(grad_func)
+
+
+class LinearizedDeformationOperator(Operator):
+
+    """Operator mapping parameters to a fixed deformed template.
+
+    This operator computes for a fixed template ``I`` the deformed
+    template::
+
+        v --> I(x + v(x))
+
+    where the vector field ``v`` depends on the deformation parameters
+    ``alpha_j`` as follows::
+
+        v(x) = sum_j (K(x, x_j) * alpha_j)
+
+    Here, ``K`` is the RKHS kernel matrix and each ``alpha_j`` is an
+    element of ``R^n``.
+    """
+
+    def __init__(self, template):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        template : `DiscreteLpVector`
+            Fixed template deformed by the vector field. Its space
+            must have the same number of dimensions as ``par_space``.
+        """
+        # The operator maps from the parameter space to the template
+        # (image) space.
+        self.template = template
+        self.domain_space = odl.ProductSpace(self.template.space,
+                                             self.template.space.ndim)
+
+        super().__init__(self.domain_space, self.template.space, linear=False)
+
+    def _call(self, displacement):
+        """Implementation of ``self(displacement)``.
+
+        Parameters
+        ----------
+        displacement: `ProductSpaceVector`
+            Linearized deformation parameters for image grid points.
+        """
+        image_pts = self.template.space.grid.points()
+        image_pts += np.asarray(displacement).T
+
+        return self.template.interpolation(image_pts.T, bounds_check=False)
+
+    def linear_deform(self, template, displacement):
```

Comment:
This method is of very limited use since it is not part of the standard `Operator` interface. If you really want the deformation operator as an operator taking both template and displacement as input, you need to make a `ProductSpaceOperator` acting on `X x V`, where `X` is the reconstruction space (where the template lives) and `V` is the space of displacement fields (usually `X^d`, where `d = dim(X)`).